### DISCUSSION TITLE
Custom AutoCompleteBox

### DISCUSSION THREAD

**Discussion by CosminCAS123** [AvaloniaUI/Avalonia#15132] [Q&A]
I am asking the user to insert his country with the help of AutoCompleteBox. Which works fine, since I just bind its ItemSource to a list of countries , but I want to customize it a bit. Besides the text, I also want to put a flag of that country in the right of the pop-up.

For example, users inputs "Ch" , Instead of just showing 2 pop-ups with China and Chad, I also want them to have their flag in the right of the pop-up. So I tried to play with the ItemTemplate of AutoCompleteBox:

	`<AutoCompleteBox HorizontalAlignment="Center"
				 Width="200"
				 Watermark="Type your country"
				 FilterMode="StartsWith"
				 ItemsSource="{Binding Countries}"
				>
            <AutoCompleteBox.ItemTemplate>
		<DataTemplate>
			<DockPanel LastChildFill="True" Margin="2" >
				<Button Content="X" DockPanel.Dock="Right"/>
				<TextBlock x:DataType="vm:SelectCountryVM"
					Text="{Binding Countries}" DockPanel.Dock="Left"/>
			</DockPanel>	
		</DataTemplate>
	</AutoCompleteBox.ItemTemplate>`
	

(the button will eventually be replaced with the image of the flag). But for now, button appears but the text doesn't. It detects it, since if I write "I" I get a result (Italy for sure), but "Italy" just isn't written , it is empty. "Countries" is a list of strings in the ViewModel with the name of the countries. Thanks!

**Comment by wsficke** [comment]
As a workaround, have you tried using an alternative structural element like Grid inside the DataTemplate?

**Reply by CosminCAS123** [reply]
This works! But I don't really understand why. {Binding} will bind the textbox's text to the whole SelectCountryVM? I don't get it .

`<AutoCompleteBox HorizontalAlignment="Center"
				 Width="200"
				 Watermark="Type your country"
				 FilterMode="StartsWith"
				 ItemsSource="{Binding Countries}"
				>
	<AutoCompleteBox.ItemTemplate>
		<DataTemplate>
			<Grid Width="200">
				<Grid.ColumnDefinitions>
					3* , *
				</Grid.ColumnDefinitions>
				<TextBox Grid.Column="0"
						 x:DataType="vm:SelectCountryVM"
						 Text="{Binding}">
				</TextBox>			
			</Grid>	
		</DataTemplate>
	</AutoCompleteBox.ItemTemplate>`

**Reply by jp2masa** [reply]
The item template is used for each item on the list. In this case, it's the country's name, so the `DataContext` inside the `DataTemplate` is a `string`.
